### PR TITLE
Headers via conan

### PR DESCRIPTION
**Pull request by partouf**
Includes https://github.com/compiler-explorer/compiler-explorer/pull/4593
Depends on https://github.com/compiler-explorer/infra/pull/975

Adds option `.packagedheaders` to libraries to indicate include files are included in the conan package. This will also trigger a download even if it's just the 'compile'-phase and not just when linking.

Also changes the directory structure from all lib files placed directly in the user's tmpdir, to [libid] and then retain the packaged directory names like `lib` and `include`.

Screenshot from 2023-06-19 20-46-57

Dir structure example:
* example.cpp
* fmt/lib/libfmtd.a
* qt/lib/libQt6Core.so
* qt/include/QtCore/QString
* etc

### CODE REVIEW

File: etc/config/c++.amazon.properties [hunk 1]
```diff
@@ -3508,6 +3508,14 @@ libs.python.versions.376.path=/opt/compiler-explorer/python-3.7.6/include/python
 libs.python.versions.381.version=3.8.1
 libs.python.versions.381.path=/opt/compiler-explorer/python-3.8.1/include/python3.8
 
+libs.qt.name=Qt
+libs.qt.url=https://www.qt.io
+libs.qt.packagedheaders=true
```
Comment (mattgodbolt) [inline etc/config/c++.amazon.properties:3513]: 🎉

**Review by mattgodbolt — APPROVED** [review]
LGTM! Would be cool (one day!) to use conan's built-in magic for this but until then...this is 👨🏻‍🍳

**Comment by partouf** [comment]
Qt has been building, doesn't support a lot of compilers, but good enough for now

**Comment by partouf** [comment]
This is now live https://godbolt.org/z/dTxsTT16d

**Pull request merged**